tests/end-to-end/verify_diff.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+"""Test different output files against expected files."""
+
+import argparse
+
+import sys
+
+import torch
+
+EPS = 0.0001
+
+
+def parse_args():
+    """Parse command line arguments.
+
+    Return:
+        args : parsed argument object.
+
+    """
+    parser = argparse.ArgumentParser(
+        "Utils scripts for verifying identicality of different files.")
+    parser.add_argument('--result_path', type=str,
+                        help='Path to generated result file.')
+    parser.add_argument('--expected_path', type=str,
+                        help='Path to expected result file.')
+    parser.add_argument('--format', type=str,
+                        choices=["torch", "text", "bw"],
+                        help='File formats being tested.')
+    parser.add_argument('--metrics', type=str,
+                        help='File formats being tested.')
+    args = parser.parse_args()
+    return args
+
+
+def verify_torch(result_path, expected_path):
+    """Verify if the two torch files are identical.
+
+    Args:
+        result_path: Path to generated result file.
+        expected_path: Path to expected result file.
+
+    """
+    rank = 0
+    result_model = torch.load(result_path, map_location="cuda:" + str(rank))
+    expected_model = torch.load(expected_path,
+                                map_location="cuda:" + str(rank))
+    for key, _ in result_model['state_dict'].items():
+        val = torch.sum((result_model['state_dict'][key] -
```

Comment:
`val` gets overwritten in each loop of the for. did you mean to use `val +=` here?

Reply:
done.